src/resolve/resolved-mdns-browse-services.c

```diff
@@ -636,19 +620,24 @@ int mdns_subscribe_browse_service(
                 .question_idna = dns_question_ref(question_idna),
                 .key = dns_question_first_key(question_utf8),
                 .ifindex = ifindex,
-                .token = token,
-                .flags = SD_RESOLVED_MDNS,
+                .flags = flags,
         };
 
-        r = sd_event_add_time(m->event,
-                        &ss->schedule_event,
-                        CLOCK_BOOTTIME,
-                        usec_add(now(CLOCK_BOOTTIME), (ss->delay * USEC_PER_SEC)),
-                        0,
-                        mdns_next_query_schedule,
-                        ss);
-        if (r < 0)
-                return r;
+        switch (flags) {
+        case SD_RESOLVED_MDNS:
```

Comment:
1.SD_RESOLVED_MDNS is available only in systemd source code, How about varlink client which uses external varlink lib.
2. flag is very generic purpose we need to extract (unmask before) using directly.
3. Overall this required some re work

Reply:
fixed